spring-boot/proxy/src/main/java/eu/coatrack/proxy/metrics/MetricsCounterService.java

```diff
@@ -52,24 +52,25 @@ public MetricsCounterService(MeterRegistry meterRegistry, MetricsTransmitter met
         this.metricsTransmitter = metricsTransmitter;
     }
 
-    public void increment(MetricsHolder mh) {
-        logBeginningOfIncrementation(mh);
-        int currentCounter = incrementAndReturnCurrentCounter(mh);
-        Metric metricToTransmit = createMetricToTransmit(mh);
-        metricToTransmit.setCount(currentCounter);
+    public void increment(TemporaryMetricsAggregation tma) {
+        logBeginningOfIncrementation(tma);
+        int currentCount = incrementAndReturnCurrentValue(tma);
+        Metric metricToTransmit = createMetricToTransmit(tma);
+        metricToTransmit.setCount(currentCount);
         metricsTransmitter.transmitToCoatRackAdmin(metricToTransmit);
     }
 
-    private void logBeginningOfIncrementation(MetricsHolder mh) {
+    private void logBeginningOfIncrementation(TemporaryMetricsAggregation tma) {
         log.debug(String.format("incrementing metric '%s' for URI '%s' and api key %s",
-                mh.getMetricType(),
-                mh.getRequestURI(),
-                mh.getApiKeyValue()
+                tma.getMetricType(),
+                tma.getRequestURI(),
+                tma.getApiKeyValue()
         ));
     }
 
-    private int incrementAndReturnCurrentCounter(MetricsHolder mh) {
-        String counterId = createCounterIdFromMetric(mh);
+    private int incrementAndReturnCurrentValue(TemporaryMetricsAggregation tma) {
```

Comment:
This sounds like the `tma` would be somehow incremented, to make it more clear I would suggest:
```suggestion
    private int incrementCounterAndReturnCurrentValue(TemporaryMetricsAggregation tma) {
```